src/VirtoCommerce.GDPR.Data/Services/DownloadContactDataService.cs

```diff
@@ -0,0 +1,90 @@
+using System.Linq;
+using System.Threading.Tasks;
+using VirtoCommerce.CustomerModule.Core.Model;
+using VirtoCommerce.CustomerModule.Core.Services;
+using VirtoCommerce.GDPR.Core.Models.DownloadData;
+using VirtoCommerce.GDPR.Core.Services;
+using VirtoCommerce.OrdersModule.Core.Model;
+using VirtoCommerce.OrdersModule.Core.Model.Search;
+using VirtoCommerce.OrdersModule.Core.Services;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.GenericCrud;
+
+namespace VirtoCommerce.GDPR.Data.Services
+{
+    public class DownloadContactDataService : IDownloadContactDataService
+    {
+        private readonly IMemberService _memberService;
+        private readonly ISearchService<CustomerOrderSearchCriteria, CustomerOrderSearchResult, CustomerOrder> _customerOrderSearchService;
+
+        public DownloadContactDataService(IMemberService memberService, ICustomerOrderSearchService customerOrderSearchService)
+        {
+            _memberService = memberService;
+            _customerOrderSearchService = (ISearchService<CustomerOrderSearchCriteria, CustomerOrderSearchResult, CustomerOrder>)customerOrderSearchService;
+        }
+
+        /// <summary>
+        /// Returns personal contact information
+        /// </summary>
+        /// <param name="id">contact Id</param>
+        /// <returns></returns>
+        public async Task<Customer> GetContactData(string id)
+        {
+            Contact contact = (Contact)await _memberService.GetByIdAsync(id);
+
+            var orderSearchCriteria = AbstractTypeFactory<CustomerOrderSearchCriteria>.TryCreateInstance();
+            orderSearchCriteria.CustomerId = contact.Id;
+            orderSearchCriteria.Take = 25;
```

Comment:
Add a comment why we take only 25 orders

Reply:
https://github.com/VirtoCommerce/vc-module-experience-api/blob/dev/src/XPurchase/VirtoCommerce.XPurchase/Services/CartProductService.cs#L30